src/runner/PlutoRunner.jl

```diff
@@ -1086,11 +1086,27 @@ function use_tree_viewer_for_struct(@nospecialize(x::T))::Bool where T
     end
 end
 
+"""
+    is_mime_enabled(::MIME) -> Bool
+
+Return whether the argument's mimetype is enabled.
+This defaults to `true`, but additional dispatches can be set to `false` by downstream packages.
+"""
+is_mime_enabled(::MIME) = true
+
+"""
+    is_tree_viewer_enabled(::MIME) -> Bool
+
+Return whether Pluto's tree viewer should be used for the argument's mimetype.
+This defaults to `true`, but additional dispatches can be set to `false` by downstream packages.
+"""
+is_tree_viewer_enabled(::MIME) = true
```

Comment:
Maybe this should be `const is_tree_viewer_enabled = Ref(true)` instead? It will only be called with `MIME"text/plain"()` as argument.

And if this is set to false, you still get Pluto's viewer for `[1,2]` and `DataFrame(...)`. Right now it just controls whether structs are rendered by Pluto's viewer, like in the test that I added.